src/parse_grammar.h

```diff
@@ -236,14 +236,28 @@ DEF_ALT(job_continuation) {
     ALT_BODY(job_continuation, piped, empty);
 };
 
-// A statement is a normal command, or an if / while / not etc.
+/// A statement like `HOME=/here emacs`
+/// We parse any number of variable_assignment nodes followed by an arbitrary statement.
+DEF(variable_assignment_prefixed_statement)
+produces_sequence<variable_assignment_list, statement>{
+    BODY(variable_assignment_prefixed_statement)};
+DEF_ALT(variable_assignment_list) {
+    using empty = grammar::empty;
+    using var = seq<variable_assignment, variable_assignment_list>;
+    ALT_BODY(variable_assignment_list, empty, var);
+};
+/// A string token like VAR=value
+DEF(variable_assignment) produces_single<tok_string>{BODY(variable_assignment)};
+
+// A statement is a normal command, or an if / while / and etc
 DEF_ALT(statement) {
     using nots = single<not_statement>;
     using block = single<block_statement>;
     using ifs = single<if_statement>;
     using switchs = single<switch_statement>;
     using decorated = single<decorated_statement>;
-    ALT_BODY(statement, nots, block, ifs, switchs, decorated);
+    using vars = seq<variable_assignment_prefixed_statement>;
```

Comment:
I think this is more recursive than necessary. It has a `statement` as perhaps containing `variable_assignment_prefixed_statement`, followed by another statement. 

I believe this would fit more naturally as part of `job` and `job_continuation` because it prevents the recursion.

    DEF(job) produces_sequence<variable_assignments, statement,  job_continuation, optional_background>{BODY(job)};

    DEF_ALT(job_continuation) {
        using piped = seq<tok_pipe, optional_newlines, variable_assignments, statement, job_continuation>;
        using empty = grammar::empty;
        ALT_BODY(job_continuation, piped, empty);
      }

Reply:
Yep, that works nicely, and the changes to the execution are smaller as well!
I've also added variable assignments to the not statement to enable something like `not a=b echo $a`.